App/Components/AsyncImage.js

```diff
@@ -0,0 +1,54 @@
+import React, { Component } from 'react'
+import { View, Image } from 'react-native'
+import IPFS from '../../TextileIPFSNativeModule'
+
+export default class AsyncImage extends React.PureComponent {
```

Comment:
This is cool, really simplifies dealing with the async image source. We could make it more general use by having one of the props just be an async function that returns the image `source`.